### DISCUSSION TITLE
V7 Conditional Controlled Component

### DISCUSSION THREAD

**Discussion by alicialics** [react-hook-form/react-hook-form#4438] [Q&A] [answered]
I am looking for v7 of https://react-hook-form.com/advanced-usage#ConditionalControlledComponent
but this isn't available in the docs.

I tried a codesandbox example and the controlled example doesnt seem to unregister.
https://codesandbox.io/s/react-hook-form-with-ui-library-ts-forked-2s4ru
Select Chocoloate for iceCream, fill in firstname, and select something else for ice cream.  firstname still exists.  v6 works correctly

**Comment by bluebill1049** [comment, marked as answer]
we are no longer auto unregister input: https://github.com/react-hook-form/react-hook-form/discussions/3714 you will have to do `unregister` at your own code. 

```
if (xxx === 'xxx') {
  unregister('input')
}
```

make sure `Controller` is removed from the view, otherwise, it will re-register again.

**Reply by bluebill1049** [reply]
you can consider upvoting this feature request as well: https://github.com/react-hook-form/react-hook-form/issues/4355
